conans/model/conf.py

```diff
@@ -1,43 +1,43 @@
-    "tools.files.download:retry_wait": "Seconds to wait between download attempts",
-    "tools.gnu:make_program": "Indicate path to make program",
-    "tools.gnu:define_libcxx11_abi": "Force definition of GLIBCXX_USE_CXX11_ABI=1 for libstdc++11",
-    "tools.google.bazel:config": "Define Bazel config file",
-    "tools.google.bazel:bazelrc_path": "Defines Bazel rc-path",
-    "tools.microsoft.msbuild:verbosity": "Verbosity level for MSBuild: "
-                                         "'Quiet', 'Minimal', 'Normal', 'Detailed', 'Diagnostic'",
-    "tools.microsoft.msbuild:vs_version": "Defines the IDE version when using the new msvc compiler",
-    "tools.microsoft.msbuild:max_cpu_count": "Argument for the /m when running msvc to build parallel projects",
-    "tools.microsoft.msbuild:installation_path": "VS install path, to avoid auto-detect via vswhere, like C:/Program Files (x86)/Microsoft Visual Studio/2019/Community",
-    "tools.microsoft.msbuilddeps:exclude_code_analysis": "Suppress MSBuild code analysis for patterns",
-    "tools.microsoft.msbuildtoolchain:compile_options": "Dictionary with MSBuild compiler options",
-    "tools.intel:installation_path": "Defines the Intel oneAPI installation root path",
-    "tools.intel:setvars_args": "Custom arguments to be passed onto the setvars.sh|bat script from Intel oneAPI",
-    "tools.system.package_manager:tool": "Default package manager tool: 'apt-get', 'yum', 'dnf', 'brew', 'pacman', 'choco', 'zypper', 'pkg' or 'pkgutil'",
-    "tools.system.package_manager:mode": "Mode for package_manager tools: 'check' or 'install'",
-    "tools.system.package_manager:sudo": "Use 'sudo' when invoking the package manager tools in Linux (False by default)",
-    "tools.system.package_manager:sudo_askpass": "Use the '-A' argument if using sudo in Linux to invoke the system package manager (False by default)",
+BUILT_IN_CONFS = {
+    "core:required_conan_version": (str, "Raise if current version does not match the defined range."),
+    "core.package_id:msvc_visual_incompatible": (str, "Allows opting-out the fallback from the new msvc compiler to the Visual Studio compiler existing binaries"),
+    "core:default_profile": (str, "Defines the default host profile ('default' by default)"),
+    "core:default_build_profile": (str, "Defines the default build profile (None by default)"),
+    "tools.android:ndk_path": (str, "Argument for the CMAKE_ANDROID_NDK"),
+    "tools.build:skip_test": (str, "Do not execute CMake.test() and Meson.test() when enabled"),
+    "tools.build:jobs": (str, "Default compile jobs number -jX Ninja, Make, /MP VS (default: max CPUs)"),
+    "tools.cmake.cmaketoolchain:generator": (str, "User defined CMake generator to use instead of default"),
+    "tools.cmake.cmaketoolchain:find_package_prefer_config": (str, "Argument for the CMAKE_FIND_PACKAGE_PREFER_CONFIG"),
+    "tools.cmake.cmaketoolchain:toolchain_file": (str, "Use other existing file rather than conan_toolchain.cmake one"),
```

Comment:
Having the types here is unexpected, and I think it makes the thing a bit unnecessarily complicated. 
Taking into account that user confs will not have the possibility to define the type, I would recommend a system that doesn't require the  type to be defined anywhere, but let the user define the type with their inputs.

Part of the surprise is that if this would be a valid feature, I would expect to see:
- ``user_toolchain`` to be a list, not str. At least to have something that is not a ``str``, otherwise, if everything is a ``str`` why defining it?
- To see tests where introducing the wrong type fails.

I am not saying to add that, I am suggesting that possibly it is not necessary, and things can be simpler.